src/SFA.DAS.ApprenticeCommitments/Data/Models/Registration.cs

```diff
@@ -58,13 +58,13 @@ public Apprentice ConvertToApprentice(string firstName, string lastName, MailAdd
         private void EnsureNotAlreadyCompleted()
         {
             if (HasBeenCompleted)
-                throw new DomainException("Already verified");
+                throw new DomainException($"Registration {Id} id already verified");
         }
 
         private void EnsureStatedEmailMatchesApproval(MailAddress emailAddress)
         {
             if (!emailAddress.ToString().Equals(Email, StringComparison.InvariantCultureIgnoreCase))
-                throw new DomainException("Email from verifying user doesn't match registered user");
+                throw new DomainException($"Email from verifying user ({emailAddress}) doesn't match registered user ({Email})");
```

Comment:
again, seek guidance on this. I don't think we should be throwing emails in exceptions because it will be logged